media/js/base/core-datalayer-init.js

```diff
@@ -5,13 +5,47 @@
 // init core dataLayer object and push into dataLayer
 $(function() {
     if (Mozilla && Mozilla.Analytics) {
-        var dataLayer = window.dataLayer = window.dataLayer || [];
+        var promises = [];
 
-        dataLayer.push({
-            'event': 'core-datalayer-loaded',
-            'pageId': Mozilla.Analytics.getPageId()
+        if (Mozilla.Client.isFirefox) {
+            var firefoxDetails = new Promise(function(resolve, reject) {
+                Mozilla.Client.getFirefoxDetails(function(details) {
+                    resolve(details);
+                });
+            });
+            promises.push(firefoxDetails);
+
+            if (/plugincheck/.test(document.location.pathname)) {
+                var pluginDetails = new Promise(function(resolve, reject) {
+                    document.addEventListener('pluginCheckDone', resolve);
+
+                    window.setTimeout(resolve, 400);
```

Comment:
Some other things I'd like to discuss in the meeting tomorrow include the robustness of doing this, over using the current GA data provided by the plugincheck page itself.